client/src/main/java/org/apache/rocketmq/client/latency/MQFaultStrategy.java

```diff
@@ -69,16 +69,17 @@ public MessageQueue selectOneMessageQueue(final TopicPublishInfo tpInfo, final S
                 }
 
                 final String notBestBroker = latencyFaultTolerance.pickOneAtLeast();
-                int writeQueueNums = tpInfo.getQueueIdByBroker(notBestBroker);
-                if (writeQueueNums > 0) {
-                    final MessageQueue mq = tpInfo.selectOneMessageQueue();
-                    if (notBestBroker != null) {
+                if (notBestBroker != null) {
+                    int writeQueueNums = tpInfo.getQueueIdByBroker(notBestBroker);
+                    if (writeQueueNums > 0) {
+                        MessageQueue mq = new MessageQueue();
+                        mq.setTopic(tpInfo.getMessageQueueList().get(0).getTopic());
```

Comment:
Why use Topic0 directly here?

Reply:
@MatrixHB This. Same confusion here. Could you please explain why directly apply topic 0 here?

Reply:
> @MatrixHB Second this. Same confusion here. Could you please explain why directly apply topic 0 here?
the topic of all messageQueue in the tpInfo.getMessageQueueList() is same,so you can choose any one,but if you get(1,2,3,4,....),may cause arrayIndexOutOfboundsException,so we get(0),if the size() ==0,it would not go into this method

Reply:
> Why use Topic0 directly here?

回复错人了我晕，我说这么久还没合进去。这个是因为  tpInfo.getMessageQueueList() 的queue全部都是同一个topic下的，所以你取这个list的哪一个元素都行，直接取0是因为代码在这个位置可以明确这个list至少有一个元素在里面，所以取0不会导致数组越界，写其他的值可能导致越界。